### PR TITLE
feat: Add non-nested callbacks

### PR DESCRIPTION
**Pull request by withgallantry**
I removed the need for nesting with callbacks. I used the commandQueue in the end. I looked at using the Enums for state but they felt like a higher level abstraction whereas the callback felt more like an implementation of the bluetooth writing characteristic. Please feel free to rewrite it to use the ENUM state instead if that feels more appropriate.

Lastly, sorry for closing the other PR, I wanted to base this off the experimental branch so that there weren't too many files in the PR.

### CODE REVIEW

**Comment by AbandonedCart** [comment]
Without the enum is perfectly fine. It was a cheap way to track the status to skip looking for the appropriate way to handle the response. It also came with the risk of taking the wrong response and running with it. 

This looks super clean. Thanks!

File: app/src/main/java/com/hiddenramblings/tagmo/BluetoothLeService.java [hunk 1]
```diff
@@ -129,7 +155,7 @@ private void getCharacteristicValue(BluetoothGattCharacteristic characteristic)
                         response = new StringBuilder();
                     }
                 } else if (progress.startsWith("tag.getList()")) {
-                    if (progress.endsWith(">")) {
+                    if (progress.endsWith(">") || progress.endsWith("\n")) {
```
Comment (AbandonedCart) [inline app/src/main/java/com/hiddenramblings/tagmo/BluetoothLeService.java:158]: The `trim` should keep this from being a scenario for `progress` but it shouldn't hurt to leave it anyway.

Comment (withgallantry) [reply app/src/main/java/com/hiddenramblings/tagmo/BluetoothLeService.java:158]: Good point. I will do some more testing and look to remove it.

**Comment by AbandonedCart** [comment]
https://github.com/HiddenRamblings/TagMo/commit/abd0661ebd5113023e5d91e06b23b4c76a19f02c should do it.

**Comment by AbandonedCart** [comment]
Random: I like the icon that replaced the loading bar on the upload. As much as I love progress indicators, the upload speed is fast enough to make the bar slow it down. The icons make it feel fancier. Nice!

**Pull request merged**